subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/incremental_check/RoomIncrementalKaptChecker.kt

```diff
@@ -0,0 +1,65 @@
+package com.avito.android.plugin.build_param_check.incremental_check
+
+import com.avito.kotlin.dsl.withType
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.internal.KaptGenerateStubsTask
+import java.net.URLClassLoader
+import javax.annotation.processing.AbstractProcessor
+
+internal class RoomIncrementalKaptChecker(
+    private val project: Project
+) {
+
+    fun isSupported(): Boolean {
+        val processor = createRoomProcessor()
+
+        try {
+            // Trying the first approach when the annotation processor is asked for supported options
+            // like it does in the real compilation process
+            return checkCommonWay(processor)
+        } catch (e: Exception) {
+            System.err.println("Fail to check Room for KAPT incremental support in official way: %e")
+        }
+
+        try {
+            return checkReflectionWay(processor)
+        } catch (e: Exception) {
+            System.err.println("Fail to check Room for KAPT incremental support in reflection way: %e")
```

Comment:
I think here must fail fast. In this case, the check is broken and must be disabled

Reply:
Fixed